### PR TITLE
QOpenGLWidget 

### PR DESCRIPTION
**Pull request by Haydelj**
Partially working implementation of the new QOpenGLWidget to replace the deprecated QGLWidget class

### CODE REVIEW

**Comment by dcwhite** [comment]
@Haydelj Time to merge master and try Windows build

**Comment by RubioJr9** [comment]
Works on Linux :confetti_ball:

File: src/Interface/Modules/Render/ES/SRInterface.h [hunk 1]
```diff
@@ -348,11 +352,17 @@ namespace SCIRun {
 
       GLuint                              mFontTexture        {};       // 2D texture for fonts
 
+<<<<<<< HEAD
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ES/SRInterface.h]: Not sure this is valid C++...

Comment (Haydelj) [reply src/Interface/Modules/Render/ES/SRInterface.h]: Hmm, what C++ version are we using?

**Comment by dcwhite** [comment]
@Haydelj Qt5+Travis is ongoing, will take a while since Travis is slow and there are even more versions to test now.

**Comment by RubioJr9** [comment]
Docking Fix:
```c++
int GuiApplication::run(int argc, const char* argv[])
{
  QApplication::setAttribute(Qt::AA_ShareOpenGLContexts);
  QApplication app(argc, const_cast<char**>(argv));
  ...
}
```

**Comment by dcwhite** [comment]
@Haydelj I got a VS state toString dump to console at one point--might want to track that down and disable it.

**Comment by dcwhite** [comment]
I'm seeing more regression test failures than before due to the screenshot not getting passed out as a matrix, but the ones I checked were still rendering fine. So this is probably the same problem with rendering/threading/state updating not all in line, but more so with the new widget structure. I'll check other systems and compare some more.

**Comment by dcwhite** [comment]
@Haydelj Just got a whiole bunch of VS state dumps--please remove those before merge can happen.

**Comment by Haydelj** [comment]
> @Haydelj Just got a whole bunch of VS state dumps--please remove those before merge can happen.

I removed it

**Comment by dcwhite** [comment]
Waiting on #1991 so we can test Travis builds

**Comment by dcwhite** [comment]
@Haydelj needs a master/qt5-dev update as we talked about

**Comment by dcwhite** [comment]
@Haydelj Ready for upstream/qt5-dev merge

**Comment by dcwhite** [comment]
Merge today into side branch

**Pull request merged**